pkg/scheduler/framework/plugins/podtopologyspread/common.go

```diff
@@ -145,6 +145,9 @@ func mergeLabelSetWithSelector(matchLabels labels.Set, s labels.Selector) labels
 }
 
 func countPodsMatchSelector(podInfos []*framework.PodInfo, selector labels.Selector, ns string) int {
+	if selector.Empty() {
```

Comment:
how about exiting here when all constraints have empty selector? https://github.com/kubernetes/kubernetes/blob/1cb334960c4288fc7830a16c97f5f8489f140b5b/pkg/scheduler/framework/plugins/podtopologyspread/scoring.go#L78

Reply:
that will save iterating over the nodes

Reply:
yes, but it changes this behavior below:

https://github.com/kubernetes/kubernetes/blob/1cb334960c4288fc7830a16c97f5f8489f140b5b/pkg/scheduler/framework/plugins/podtopologyspread/scoring.go#L81

Reply:
Note that if you set a DoNotSchedule constraint, then we also don't apply default constraints in scoring. In that case the list of constraints in PreScore is zero.